compiler/src/iree/compiler/Codegen/LLVMGPU/Utils/LLVMGPULayoutAnalysisAndDistribution.cpp

```diff
@@ -1281,6 +1282,31 @@ static bool isMatmulTransposeB(vector::ContractionOp contractOp) {
 
 void doLayoutAnalysisAndDistribution(RewriterBase &rewriter,
                                      func::FuncOp funcOp) {
+  DataFlowSolver solver;
+  // The order of loading matters here, because propagateLayout does some
+  // initialization which needs the lattice to know both enforcement and
+  // propagation.
```

Comment:
Something worth considering is whether we would need to run enforcement and propagation for a fixed number of iterations till the layouts "converge".